src/control/server/ctl_storage_rpc.go

```diff
@@ -306,3 +306,59 @@ func (c *ControlService) StorageNvmeRebind(ctx context.Context, req *ctlpb.NvmeR
 
 	return resp, nil
 }
+
+// StorageNvmeAddDevice rebinds SSD from kernel and binds to user-space to allow DAOS to use it.
+func (c *ControlService) StorageNvmeAddDevice(ctx context.Context, req *ctlpb.NvmeAddDeviceReq) (resp *ctlpb.NvmeAddDeviceResp, err error) {
+	c.log.Debugf("received StorageNvmeAddDevice RPC %v", req)
+
+	if req == nil {
+		return nil, errors.New("nil request")
+	}
+
+	instances := c.harness.Instances()
+	engineIndex := req.GetEngineIndex()
+	if len(instances) <= int(engineIndex) {
+		return nil, errors.Errorf("engine with index %d not found", engineIndex)
+	}
+	defer func() {
+		err = errors.Wrapf(err, "engine %d", engineIndex)
+	}()
+
+	engine := instances[engineIndex]
+	ei, ok := engine.(*EngineInstance)
+	if !ok {
+		return nil, errors.New("not an EngineInstance")
+	}
```

Comment:
It occurs to me that maybe it would be better to add a GetStorage method to the Engine interface to make it possible to get the `ei.storage` member, rather than having to do this type check. I know we're trying to slim down that interface, but having a general storage provider seems like a relevant idea to the DAOS Engine in general.

Reply:
sounds reasonable to me, I will make the change